### ISSUE TITLE
`RegisterKeyMapping` Native doesn't exist for `rdr3` / RedM

### ISSUE THREAD

**Issue by Z3rio** [citizenfx/fivem#1779]
The native `RegisterKeyMapping` seems to exist both ingame and in the native list, although it does nothing at all upon being used.
No errors at all happen or such, which is why I would believe it exists but isnt compatible with RedM.

I would guess it was directly copied from FiveM and therefore might not work with RedM.
I tried to take a look at this myself, although I'm not that efficient in c++ etc.

Would greatly **appreciate** it if someone could take a look at this since its quite a dealbreaker.

**Comment by Disquse** [comment]
> The native RegisterKeyMapping seems to exist both ingame

It doesn't exist in-game (in RedM).

> in the native list,

And it's not in the natives list. You probably confused FiveM official reference and unofficial RedM reference.

> Would greatly appreciate it if someone could take a look at this since its quite a dealbreaker.

For now this is a low-priority task.

**Comment by Z3rio** [comment]
> And it's not in the natives list. You probably confused FiveM official reference and unofficial RedM reference.
Yeah, I guess I might have done that then, do you perhaps have the link to RedM's native list?

> For now this is a low-priority task.
Why? Dont you think it creates a lot of unneccesary resource usage to have a bunch of while loops checking it every millisecond instead? (I'm talking about stuff that has to be checked 24/7, not only at specific moments)

**Comment by Disquse** [comment]
> Why? Dont you think it creates a lot of unneccesary resource usage to have a bunch of while loops checking it every millisecond instead? (I'm talking about stuff that has to be checked 24/7, not only at specific moments)

Because RedM developers are busy with more important stuff like fixing OneSync issues and crashes. Key mapping support is planned and might appear later this year. However, this is not an easy task, it would require to port a lot of code from FiveM and research the changes that were made by R* in RDR3 comparing to GTA5.

**Comment by Z3rio** [comment]
> > Why? Dont you think it creates a lot of unneccesary resource usage to have a bunch of while loops checking it every millisecond instead? (I'm talking about stuff that has to be checked 24/7, not only at specific moments)
> 
> Because RedM developers are busy with more important stuff like fixing OneSync issues and crashes. Key mapping support is planned and might appear later this year. However, this is not an easy task, it would require to port a lot of code from FiveM and research the changes that were made by R* in RDR3 comparing to GTA5.
> 
> > Yeah, I guess I might have done that then, do you perhaps have the link to RedM's native list?
> 
> There's no official list yet. You can check native declarations in `citizen/scripting/*` files. This probably going to be fixed soon.

I would love to help in any way that I can.
Do you know if the Onesync issues are FiveM sided // C++
Or if they are in the actual lua resources being ran on RedM

**Comment by Disquse** [comment]
> Do you know if the Onesync issues are FiveM sided // C++

No, this is about fixing OneSync issues on RedM. FiveM OneSync works well.

**Comment by Z3rio** [comment]
> > Do you know if the Onesync issues are FiveM sided // C++
> 
> No, this is about fixing OneSync issues on RedM. FiveM's OneSync works well.

Ah my bad, I meant RedM mate. I'm just a sloppy typer haha.
I meant if the issues with RedM's Onesync is in the actual src files // the files that are in this repo.
Or if the issues are caused by other scripts misusing Onesync.

**Comment by outsider31000** [comment]
another way is to use the new implementation made by @AvarianKnight
https://github.com/citizenfx/fivem/blob/master/ext/native-decls/RegisterRawKeymap.md 
not sure if with this RegisterKeyMapping is needed?
